### PR TITLE
Update kubeadmin password for each fresh start

### PR DESCRIPTION
**Pull request by praveenkumar**
This patch will create a random password and then update the cluster
secret for htpasswd. It also update the cache kubeadmin-password file
so that all the commands which rely on it doesn't break.

fixes: #2048

### CODE REVIEW

File: pkg/crc/cluster/kubeadminPassword.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	"github.com/code-ready/crc/pkg/crc/machine/bundle"
+	"github.com/code-ready/crc/pkg/crc/oc"
+	"golang.org/x/crypto/bcrypt"
+)
+
+// UpdateKubeAdminUserPassword does following
+// - Generate a random password for kubeadmin user
+// - Update the kubeadmin password file
+// - Update the htpasswd secret
+func UpdateKubeAdminUserPassword(kubeAdminPassword string, ocConfig oc.Config, bundle *bundle.CrcBundleInfo) error {
```
Comment (guillaumerose) [inline pkg/crc/cluster/kubeadminPassword.go:18]: Is it possible to create a new file with the password in ~/.crc/machines/crc/password ? I don't really like changing a file in the cache after start. It is like "corrupting" it.

File: pkg/crc/cluster/kubeadminPassword.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	// In case of cluster started from stopped state.
+	if kubeAdminPassword == "" {
+		return nil
+	}
+	hashDeveloperPasswd, err := hashBcrypt("developer")
```
Comment (guillaumerose) [inline pkg/crc/cluster/kubeadminPassword.go:23]: developer user is sudoer. I guess we should also rotate the password. WDYT?

Comment (praveenkumar) [reply pkg/crc/cluster/kubeadminPassword.go:23]: we can remove developer from sudoer.

**Comment by praveenkumar** [comment]
I am thinking why we have any password set from snc side? We do have kubecofig file for `system:admin` role and same is consumed by all the `oc` commands. We can consume same kubeconfig file for creating <user> - <password> from crc side?

**Comment by guillaumerose** [comment]
Yes we can remove the kubeadmin file from the bundle :+1:

File: pkg/crc/cluster/kubeadminPassword.go (hunk 2 above)
Comment (guillaumerose) [reply pkg/crc/cluster/kubeadminPassword.go:22]: Yes but we need to also to update the doc to remove this kind of commands: `oc get co --as system:admin`.

File: pkg/crc/machine/bundle/metadata.go [hunk 3]
```diff
@@ -111,13 +111,27 @@ func (bundle *CrcBundleInfo) GetInitramfsPath() string {
 }
 
 func (bundle *CrcBundleInfo) GetKubeadminPassword() (string, error) {
```
Comment (guillaumerose) [inline pkg/crc/machine/bundle/metadata.go:113]: You can delete this function in fact now. kubeadmin password in bundle will disppear. It removes the if in ~/.crc/machines/... below.

Comment (guillaumerose) [reply pkg/crc/machine/bundle/metadata.go:113]: Ok :+1:

**Pull request merged**